app/controllers/HasNirmsController.scala

```diff
@@ -103,8 +103,11 @@ class HasNirmsController @Inject() (
                 updatedAnswers                  <- Future.fromTry(request.userAnswers.set(HasNirmsUpdatePage, value))
                 updatedAnswersWithTraderProfile <-
                   Future.fromTry(updatedAnswers.set(TraderProfileQuery, traderProfile))
-                _                               <- sessionRepository.set(updatedAnswersWithTraderProfile)
-              } yield Redirect(navigator.nextPage(HasNirmsUpdatePage, mode, updatedAnswersWithTraderProfile))
+                updatedAnswersWithRemoveNirms <- Option.when(traderProfile.nirmsNumber.isEmpty && !value) {
+                  Future.fromTry(updatedAnswersWithTraderProfile.set(RemoveNirmsPage, true))
+                }.getOrElse(Future.successful(updatedAnswersWithTraderProfile))
+                _                               <- sessionRepository.set(updatedAnswersWithRemoveNirms)
+              } yield Redirect(navigator.nextPage(HasNirmsUpdatePage, mode, updatedAnswersWithRemoveNirms))
 
```

Comment:
The above is logically equivalent to this (I was just experimenting with scala): 

```
if (traderProfile.nirmsNumber.isEmpty && !value) {
  Future.fromTry(updatedAnswersWithTraderProfile.set(RemoveNirmsPage, true))
} else {
  Future.successful(updatedAnswersWithTraderProfile)
}
```